editoast/src/modelsv2/projects.rs

```diff
@@ -0,0 +1,241 @@
+    #[diesel(sql_type = BigInt)]
+    pub studies_count: i64,
+}
+
+impl ProjectWithStudies {
+    pub fn new_from_project(project: Project, studies_count: i64) -> Self {
+        Self {
+            project: ProjectRow {
+                id: project.id,
+                name: project.name,
+                objectives: project.objectives,
+                description: project.description,
+                funders: project.funders,
+                budget: project.budget,
+                creation_date: project.creation_date,
+                last_modification: project.last_modification,
+                tags: project.tags,
+                image: project.image,
+            },
+            studies_count,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Default, ToSchema)]
+pub enum Ordering {
+    NameAsc,
+    NameDesc,
+    CreationDateAsc,
+    CreationDateDesc,
+    #[default]
+    LastModifiedDesc,
+    LastModifiedAsc,
+}
+
+impl Ordering {
+    pub fn to_sql(&self) -> &str {
+        match *self {
+            Ordering::NameAsc => "LOWER(t.name) ASC",
+            Ordering::NameDesc => " LOWER(t.name) DESC",
+            Ordering::CreationDateAsc => "creation_date",
+            Ordering::CreationDateDesc => "creation_date DESC",
+            Ordering::LastModifiedAsc => "last_modification",
+            Ordering::LastModifiedDesc => "last_modification DESC",
+        }
+    }
+}
+
+impl Project {
+    /// This function takes a filled project and update to now the last_modification field
+    pub async fn update_last_modified(self, db_pool: Data<DbPool>) -> Result<Option<Project>> {
+        let mut conn = db_pool.get().await?;
+        self.update_last_modified_conn(&mut conn).await
+    }
+
+    pub async fn update_last_modified_conn(
+        mut self,
+        conn: &mut PgConnection,
+    ) -> Result<Option<Project>> {
+        self.last_modification = Utc::now().naive_utc();
+        self.save(conn).await?;
+        Ok(Some(self))
+    }
+
+    pub async fn with_studies(self, db_pool: Data<DbPool>) -> Result<ProjectWithStudies> {
+        use crate::tables::study::dsl as study_dsl;
+        let mut conn = db_pool.get().await?;
+        let studies_count = study_dsl::study
+            .filter(study_dsl::project_id.eq(self.id))
+            .count()
+            .get_result(&mut conn)
+            .await?;
+        Ok(ProjectWithStudies {
+            project: ProjectRow {
+                id: self.id,
+                name: self.name,
+                objectives: self.objectives,
+                description: self.description,
+                funders: self.funders,
+                budget: self.budget,
+                creation_date: self.creation_date,
+                last_modification: self.last_modification,
+                tags: self.tags,
+                image: self.image,
+            },
+            studies_count,
+        })
+    }
+}
+
+#[async_trait]
+impl List<Ordering> for ProjectWithStudies {
```

Comment:
imo:

```suggestion
impl List<Ordering> for Project {
```

and let the views fill in the study count

Reply:
```
#[async_trait]
impl List<Ordering> for ProjectWithStudies {
    async fn list_conn(
        conn: &mut PgConnection,
        page: i64,
        page_size: i64,
        ordering: Ordering,
    ) -> Result<PaginatedResponse<Self>> {
        let ordering = ordering.to_sql();
        sql_query(format!(
            "SELECT t.*, COUNT(study.*) as studies_count FROM project as t
            LEFT JOIN study ON study.project_id = t.id
            GROUP BY t.id ORDER BY {ordering}"
        ))
        .paginate(page, page_size)
        .load_and_count(conn)
        .await
    }
}
```
Do you mean to search for a list of projects only, without including any studies?
If so, we'll need to perform a separate query for each project to retrieve its associated studies.